CustomFeeValidator.cs

```diff
@@ -70,14 +91,21 @@ public static float GetTotalDebtForJobPurposes(DisplayableDebt debt)
             if (locoInTrainset != null)
             {
                 float fees = debt.GetTotalPriceOfResources(nonConsumableTypes);
-                DebugLog($"locomotive {locoInTrainset.ID} is in player's last trainset. Fees without consumables = {fees}");
+                DebugLog($"Locomotive {locoInTrainset.ID} is in player's last trainset. Fees without consumables = {fees}.");
                 return fees;
             }
-            else
-            {
-                DebugLog($"locomotive {debt.ID} not found in last trainset");
+            DebugLog($"Locomotive {debt.ID} not found in last trainset.");
+            return debt.GetTotalPrice();
+        }
+
+        private static float GetTotalDebtForFeeValidationTypeExistingLocos(DisplayableDebt debt)
+		{
+            ExistingLocoDebt locoDebt = debt as ExistingLocoDebt;

```

Comment:
You can use `if (debt is ExistingLocoDebt)` since locoDebt isn't actually used for anything.

The whole method could be:
`return debt is ExistingLocoDebt ? 0 : debt:GetTotalPrice();` 😉

Reply:
I refactored this a bit, but not as far as to make it a single line return. This way we can keep the logging around for the Debug build.